src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animals) {
+    if (!animals.hidden && animals instanceof Herbivore) {
+      animals.health -= 50;
+    }
+
+    if (animals.health <= 0) {
+      Animal.alive.splice(Animal.alive.indexOf(animals), 1);
```

Comment:
from checklist
2. [CODE PERFOMANCE] - Try to avoid nesting loops (splice  + indexof), use filter instead

Reply:
Ur right thank you, i will try to fix that asap.